src/Assignments/ProblemIframe.tsx

```diff
@@ -92,6 +122,8 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
 
         let formData = new FormData(problemForm);
         if (!_.isNil(clickedButton)) {
+            // current state will never match submission unless we save it before including clickedButton
+            setLastSubmission(formDataToObject(formData)); 
```

Comment:
This will occur on all button clicks including `return to default`, we should explicitly check for submit button